packages/horizontal-layout/src/vaadin-horizontal-layout-styles.js

```diff
@@ -27,4 +29,18 @@ export const horizontalLayoutStyles = css`
   :host([theme~='spacing']) {
     gap: 1em;
   }
+
+  ${enableLayoutImprovements
+    ? unsafeCSS`
+    ::slotted([style^='width: 100%']),
+    ::slotted([style*=' width: 100%']) {
+      flex: 1;
+    }
+  
+    ::slotted(vaadin-horizontal-layout),
+    ::slotted(vaadin-vertical-layout) {
+      min-width: 0;
+    }
```

Comment:
Similarly, changing the min size on all children is not desirable. For example, for grid we explicitly added a min-height in the shadow root styles, which should not be overridden by the layout. Other components have an explicit min size as well. As the problem with children not being able to shrink due to their content mostly involves nested layouts we chose to limit this to these two components for now.